### ISSUE TITLE
FIX: Weird characters in text of csv file

### ISSUE THREAD

**Issue by bmvoss** [DataKind-DC/capital-nature-ingest#55]
## Current Behavior
It looks like the csv files are showing special characters with weird symbols:
- apostrophes (') shown as â€™
- quotation marks (") shown as â€œ or â€
- hyphens (-) shown as â€“
- colons (:) shown as :Â
Not _every_ instance of these characters is replaced with these weird strings, but a lot are (a few hundred in total). There are also seemingly random Â symbols (by themselves) all over the file.

## Possible Solution
I don't know!

## Steps to Reproduce (for bugs)
See the csv file produced on 2/19/19.

## Context
Want to remove these weird characters before publishing events to the calendar. Hopefully it's possible to eliminate them in the code (rather than doing find-and-replace in Excel).

**Comment by kirakrishnan** [comment]
@csmcallister  I am working on enh #53 before concatenating the csv if I run it thru unicodedata.normalize('NFKD',s) or `s = s.encode('ascii',errors='ignore')` will it help? for this

**Comment by csmcallister** [comment]
@kirakrishnan Yeah, this is likely an encoding issue, so that might help. You could also specify an encoding when writing to csv. We probably need something like latin1. I'll DM you all of the separate csvs so you can give it a shot.

**Comment by csmcallister** [comment]
Using the this `unicode_normalizer` function with the `applymap` method in pandas fixes the encoding issue:

```python
import pandas as pd

files = ['vnps-results.csv',
         'ans-results.csv',
         'arlington-results.csv',
         'casey_trees-results.csv',
         'fairfax-results.csv',
         'montgomery-results.csv',
         'nps-results.csv']

def unicode_normalizer(x):
    if isinstance(x, str):
        x = x.encode('windows-1252').decode("utf8", errors='ignore')
        return x
    else:
        return x

dfs = []
for f in files:
    df = pd.read_csv(f)
    df = df.applymap(unicode_normalizer)
    dfs.append(df)

final_df = pd.concat(dfs, sort = True)

final_df.to_csv('cap-nature-events-021919.csv', 
                          index=False)
```

For a long term solution, we'll want to use this where the csvs are first written. More on that to come!

**Comment by csmcallister** [comment]
Ok, I added something to address this and wrote some tests to ensure it works. If we notice new things, we can note them and then I can update the tests to make sure the solution works.

**Comment by csmcallister** [comment]
closed by c93faf9
